### PR TITLE
BlackBox specs #2524

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Refactor
- [x] Bug Fix

## Description
- added specs for `BlackBox` and `FunctionCaller`
- created an `Aws::FakeClient` not to make actual calls to the aws lambda in non-production enviroments
I could use webmock or VCR to mock calls in the test environment, but the actual calls shouldn't be done in the dev environment as well.
- fixed the bug when `nil` was returned from the aws lambda function
- this pr also fixes the opening the `/settings/membership` page being logged in as the sustaining member in the dev environment

## Related Tickets & Documents
#2524

### CODE REVIEW

File: app/services/aws/fake_client.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Fake Aws client is used in non-production environments to prevent actual calls to AWS lambda
+module Aws
+  class FakeClient
+    def initialize; end
```
Comment (rhymes) [inline app/services/aws/fake_client.rb]: isn't this superfluous?

File: spec/black_box/black_box_spec.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      score = described_class.article_hotness_score(article, function_caller)
+      expect(score).to eq(1298)
+    end
+  end
+
+  describe "#comment_quality_score" do
+    let(:comment) { create(:comment, commentable: create(:article), body_markdown: "```#{'hello, world! ' * 20}```") }
+
+    before do
+      reaction = create(:reaction, reactable: comment)
```
Comment (rhymes) [inline spec/black_box/black_box_spec.rb]: you can just say `create(:reaction, reactable: comment, points: 20)` I guess

File: spec/labor/coupon_generator_spec.rb [hunk 3]
```diff
@@ -5,10 +5,8 @@
 
   describe "#expect" do
     it "generates code with proper prefix ( [version]_[code] )" do
-      VCR.use_cassette "coupon_generator_1" do
-        version = versions.sample
-        expect(described_class.new(1, version).generate).to include("#{version}_")
-      end
+      version = versions.sample
```
Comment (rhymes) [inline spec/labor/coupon_generator_spec.rb]: I think you can delete `spec/support/fixtures/vcr_cassettes/coupon_generator_1.yml` if it's not needed

File: spec/black_box/black_box_spec.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      reaction.update_column(:points, 20)
```
Comment (rhymes) [inline spec/black_box/black_box_spec.rb]: you can just say create(:reaction, reactable: comment, points: 20) I guess

Comment (lightalloy) [reply spec/black_box/black_box_spec.rb]: points will be assigned in the `before_save` callback then and the specified value will be overwritten

**Pull request merged**